File: app/src/main/java/dev/ragnarok/fenrir/api/model/response/ItemsProfilesGroupsResponse.java
package dev.ragnarok.fenrir.api.model.response;

import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import dev.ragnarok.fenrir.api.model.VKApiCommunity;
import dev.ragnarok.fenrir.api.model.VKApiUser;

public class ItemsProfilesGroupsResponse<T> {

    @Nullable
    @SerializedName("items")
    public List<T> items;

    @SerializedName("count")
    public int count;

    @Nullable
    @SerializedName("profiles")
    public List<VKApiUser> profiles;

    @Nullable
    @SerializedName("groups")
    public List<VKApiCommunity> groups;
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/model/response/ItemsProfilesGroupsResponse.kt
package dev.ragnarok.fenrir.api.model.response

import com.google.gson.annotations.SerializedName
import dev.ragnarok.fenrir.api.model.VKApiCommunity
import dev.ragnarok.fenrir.api.model.VKApiConversation
import dev.ragnarok.fenrir.api.model.VKApiUser

class ItemsProfilesGroupsResponse<T> {
    @SerializedName("items")
    var items: List<T>? = null

    @SerializedName("count")
    var count = 0

    @SerializedName("profiles")
    var profiles: List<VKApiUser>? = null

    @SerializedName("groups")
    var groups: List<VKApiCommunity>? = null

    @SerializedName("contacts")
    var contacts: List<VKApiConversation.ContactElement>? = null
}